tests/unit/test_routes.py

```diff
@@ -9,54 +9,231 @@
+    client = module_client
+
+    response = client.post('api/v1/apikey', json=dict(
+        email="test@example.org",
+        password="supersecurepassword"
+    ))
+
+    assert (response.status_code == 200)
+    assert (response.json['data'].get('email') == "test@example.org")
+    assert (isinstance(response.json['data'].get('apikey'), str))
+
+
+def test_create_resource(module_client, module_db, patched_request):
+    client = module_client
+
+    apikey = get_api_key(client)
+    response = create_resource(client, apikey)
     assert (response.status_code == 200)
     assert (isinstance(response.json['data'].get('id'), int))
     assert (response.json['data'].get('name') == "Some Name")
 
 
-def test_update_resource(session_app, function_session, session_db):
-    client = session_app.test_client()
+def test_update_resource(module_client, module_db, patched_request):
+    client = module_client
 
-    response = create_resource(client)
+    apikey = get_api_key(client)
 
-    id = response.json['data'].get('id')
-    assert (isinstance(id, int))
-
-    response = client.put(f"/api/v1/resources/{id}", json={
-        "name": "New name"
-    })
+    response = client.put("/api/v1/resources/1",
+        json={ "name": "New name" },
+        headers={ 'x-apikey': apikey }
+    )
 
     assert (response.status_code == 200)
     assert (response.json['data'].get('name') == "New name")
 
 
-def test_rate_limit(session_app, function_session, session_db):
-    client = session_app.test_client()
+##########################################
+## Other Routes
+##########################################
+
+
+# This method must come last if using the persistent client and db
+def test_rate_limit(module_client, module_db):
+    client = module_client
 
     for _ in range(50):
         client.get('api/v1/resources')
 
     # Response should be a failure on request 51
     response = client.get('api/v1/resources')
     assert(response.status_code == 429)
-    assert(type(response.json.get('errors')) is list)
+    assert (isinstance(response.json.get('errors'), list))
     assert(response.json.get('errors')[0].get('code') == "rate-limit-exceeded")
 
 
 ##########################################
 ## Helpers
 ##########################################
-def create_resource(client):
-    return client.post('/api/v1/resources', json={
-        "name": "Some Name",
-        "url": "http://example.org/",
-        "category": "New Category",
-        "languages": ["Python", "New Language"],
-        "paid": False,
-        "notes": "Some notes"
-    })
+def create_resource(client, apikey):
+    return client.post('/api/v1/resources',
+        json={
+            "name": "Some Name",
+            "url": "http://example.org/",
+            "category": "New Category",
+            "languages": ["Python", "New Language"],
+            "paid": False,
+            "notes": "Some notes"
+        },
+        headers={'x-apikey': apikey}
```

Comment:
These ones need to be in this format because if you do this
```python
headers = dict(x-apikey=apikey)
```
you get this:
```SyntaxError: keyword can't be an expression```